clients/drcachesim/reader/reader.cpp

```diff
@@ -204,8 +204,12 @@ reader_t::process_input_entry()
             // Look for encoding bits that belong to this instr.
```

Comment:
Is this same as the cause of i#6303? So we don't have a fix yet?

Reply:
> You're asking whether the record-filter-caused encoding problems like #6712 are related to #6303? 

This PR also adds an ERRMSG in reader code, so wasn't sure if there's any connection to #6303. Thanks for clarifying.

Reply:
The reader ERRMSG was one way we detected a problem.  It is now a release-build abort so it should help us find such problems sooner in the future instead of days later in this case where we wasted time thinking things were fine.